i18n_taxonomy/i18n_taxonomy.install

```diff
@@ -75,9 +68,70 @@ function i18n_taxonomy_disable() {
   }
 }
 
+/**
+ * Implements hook_update_dependencies().
+ */
+function i18n_taxonomy_update_dependencies() {
+  // Set the i18n_mode for vocabularies between taxonomy_update_1001 and
+  // taxonomy_update_1002
+  $dependencies['i18n_taxonomy'][1000] = array(
+    'taxonomy' => 1001,
+  );
+  $dependencies['taxonomy'][1002] = array(
+    'i18n_taxonomy' => 1000,
+  );
+  // Squeeze i18n_taxonomy_update_1001 in between taxonomy_update_1006 and
+  // taxonomy_update_1007.
+  $dependencies['i18n_taxonomy'][1001] = array(
+    'taxonomy' => 1006,
+  );
+  $dependencies['taxonomy'][1007] = array(
+    'i18n_taxonomy' => 1001,
+  );
+
+  return $dependencies;
+}
+
 /**
  * Implements hook_update_last_removed().
  */
 function i18n_taxonomy_update_last_removed() {
   return 7004;
 }
+
+/**
+ * Set the i18n mode per vocabulary.
+ */
+function i18n_taxonomy_update_1000() {
+  if (!db_table_exists('taxonomy_vocabulary')) {
+    return;
+  }
+  $result = db_query('SELECT machine_name, language, i18n_mode FROM {taxonomy_vocabulary}');
```

Comment:
Should we also check that `i18n_mode` field exists?

Reply:
Then we probably should check for both columns, language and i18n_mode. Both get added by i18n_taxonomy.